### ISSUE TITLE
Node not getting blocks until after a few hours

### ISSUE THREAD

**Issue by NotToFly** [input-output-hk/jormungandr#995]
After bootstarp stuck on connecting to peer. no updates no errors. Rust and Nix node the same.

**Mandatory Information**
1. jcli 0.6.3 (master-bb925b7a+, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)
2. jormungandr 0.6.3 (master-bb925b7a+, release, linux [x86_64]) - [rustc 1.38.0 (625451e37 2019-09-23)

**To Reproduce**
1. Start Jormungandr with
jormungandr --genesis-block-hash adbdd5ede31637f6c9bad5c271eec0bc3d0cb9efb86a5b913bb55cba549d0770 --config node-config.yaml --secret node-secret.yaml --log-level=info

Not the last release related. was like that for a while.

![19-Oct-19 12-05-07 PM](https://user-images.githubusercontent.com/53143161/67144066-7848c180-f26a-11e9-9c22-c2d84148a3ad.png)

**Comment by mzabaluev** [comment]
Please update to 0.6.5 and see if the problem persists.

**Comment by jbax** [comment]
I'm having the same issue here with 0.6.5, right now `uptime: 3183` and no blocks yet. Usually it starts syncing after 4 to 6 hours - it has been like that since version 0.5.4 which is the version I began testing with.

**Comment by mzabaluev** [comment]
Setting log level to `debug` would enable dumping of protocol messages received by the node (among gobs of other information). Please check if the node receives any kind of block events, provided that it's even connected to any other nodes.

**Comment by jbax** [comment]
Are there any messages in specific from the log that I should look for?

**Comment by NotToFly** [comment]
Updated to 0.6.5. 
Uptime 7200 
Debug screen below

![19-Oct-19 6-33-37 PM](https://user-images.githubusercontent.com/53143161/67149122-49007780-f29f-11e9-922b-35120bb0fe0a.png)

**Comment by jbax** [comment]
My uptime is at 31645 right now. The log shows what you are looking for:

```
Oct 20 08:43:53.931 DEBG scheduling Read for: 0, scope: global
Oct 20 08:43:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:43:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:43:53.931 DEBG scheduling Read for: 0, scope: global
Oct 20 08:43:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:43:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:03.930 DEBG selecting gossips for 91672571fddfd0a3d95e0b955b8f2f62c818a32f35c1ab97667033c1d4fcaea3, task: network
Oct 20 08:44:03.930 DEBG sending gossip to node 91672571fddfd0a3d95e0b955b8f2f62c818a32f35c1ab97667033c1d4fcaea3, task: network
Oct 20 08:44:03.930 DEBG selecting gossips for a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 20 08:44:03.930 DEBG sending gossip to node a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 20 08:44:03.930 DEBG selecting gossips for d05ca82e8e41ab4a1f1a5e7723ac4c784188f800904948a60b6904e374816649, task: network
Oct 20 08:44:03.930 DEBG sending gossip to node d05ca82e8e41ab4a1f1a5e7723ac4c784188f800904948a60b6904e374816649, task: network
Oct 20 08:44:03.930 DEBG selecting gossips for e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:03.930 DEBG sending gossip to node e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:03.930 DEBG scheduling Read for: 1, scope: global
Oct 20 08:44:03.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:03.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:03.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:03.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:03.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:03.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:03.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:13.930 DEBG selecting gossips for 657127e5a8d84e956e88c98f4923e38287c672a2f19043a5971bf7d57f64bac9, task: network
Oct 20 08:44:13.930 DEBG sending gossip to node 657127e5a8d84e956e88c98f4923e38287c672a2f19043a5971bf7d57f64bac9, task: network
Oct 20 08:44:13.930 DEBG selecting gossips for 9861dfd6450967f4c1e986d32223a5e02d44621f27186268393186f946b23c47, task: network
Oct 20 08:44:13.930 DEBG sending gossip to node 9861dfd6450967f4c1e986d32223a5e02d44621f27186268393186f946b23c47, task: network
Oct 20 08:44:13.930 DEBG selecting gossips for a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 20 08:44:13.930 DEBG sending gossip to node a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 20 08:44:13.930 DEBG selecting gossips for e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:13.930 DEBG sending gossip to node e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:13.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:13.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:13.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:13.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:13.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:13.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:13.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:13.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:23.929 DEBG selecting gossips for 657127e5a8d84e956e88c98f4923e38287c672a2f19043a5971bf7d57f64bac9, task: network
Oct 20 08:44:23.929 DEBG sending gossip to node 657127e5a8d84e956e88c98f4923e38287c672a2f19043a5971bf7d57f64bac9, task: network
Oct 20 08:44:23.929 DEBG selecting gossips for 7a36878ac60f05645cd889671eeaba0b47cd1b0aa8f1729fd843bc20503b22f7, task: network
Oct 20 08:44:23.929 DEBG sending gossip to node 7a36878ac60f05645cd889671eeaba0b47cd1b0aa8f1729fd843bc20503b22f7, task: network
Oct 20 08:44:23.929 DEBG selecting gossips for 9861dfd6450967f4c1e986d32223a5e02d44621f27186268393186f946b23c47, task: network
Oct 20 08:44:23.929 DEBG sending gossip to node 9861dfd6450967f4c1e986d32223a5e02d44621f27186268393186f946b23c47, task: network
Oct 20 08:44:23.929 DEBG selecting gossips for e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:23.929 DEBG sending gossip to node e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:23.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:23.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:23.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:23.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:23.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:23.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:23.930 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:23.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:33.932 DEBG selecting gossips for 657127e5a8d84e956e88c98f4923e38287c672a2f19043a5971bf7d57f64bac9, task: network
Oct 20 08:44:33.932 DEBG sending gossip to node 657127e5a8d84e956e88c98f4923e38287c672a2f19043a5971bf7d57f64bac9, task: network
Oct 20 08:44:33.933 DEBG selecting gossips for 91672571fddfd0a3d95e0b955b8f2f62c818a32f35c1ab97667033c1d4fcaea3, task: network
Oct 20 08:44:33.933 DEBG sending gossip to node 91672571fddfd0a3d95e0b955b8f2f62c818a32f35c1ab97667033c1d4fcaea3, task: network
Oct 20 08:44:33.933 DEBG selecting gossips for a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 20 08:44:33.933 DEBG sending gossip to node a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 20 08:44:33.933 DEBG selecting gossips for e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:33.933 DEBG sending gossip to node e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 20 08:44:33.933 DEBG scheduling Read for: 1, scope: global
Oct 20 08:44:33.933 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:33.933 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:33.933 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:33.933 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:33.933 DEBG scheduling Read for: 0, scope: global
Oct 20 08:44:33.933 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 20 08:44:33.933 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
```

Yet, the node is not syncing:

```
blockRecvCnt: 0
lastBlockDate: "238.41974"
lastBlockFees: 0
lastBlockHash: 5ec4db50e6e55ae78e8ea3476adc4d06b84736d126df89225a40bd55ea232137
lastBlockHeight: "151508"
lastBlockSum: 0
lastBlockTime: ~
lastBlockTx: 0
txRecvCnt: 0
uptime: 31645

```

**Comment by jbax** [comment]
Update: my node is still running and connected to the other nodes:

```
Oct 21 06:41:33.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:33.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:33.930 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:33.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:43.929 DEBG selecting gossips for 7a36878ac60f05645cd889671eeaba0b47cd1b0aa8f1729fd843bc20503b22f7, task: network
Oct 21 06:41:43.929 DEBG sending gossip to node 7a36878ac60f05645cd889671eeaba0b47cd1b0aa8f1729fd843bc20503b22f7, task: network
Oct 21 06:41:43.929 DEBG selecting gossips for a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 21 06:41:43.930 DEBG sending gossip to node a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 21 06:41:43.930 DEBG selecting gossips for d05ca82e8e41ab4a1f1a5e7723ac4c784188f800904948a60b6904e374816649, task: network
Oct 21 06:41:43.930 DEBG sending gossip to node d05ca82e8e41ab4a1f1a5e7723ac4c784188f800904948a60b6904e374816649, task: network
Oct 21 06:41:43.930 DEBG selecting gossips for e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 21 06:41:43.930 DEBG sending gossip to node e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 21 06:41:43.930 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:43.930 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:43.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:43.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:43.930 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:43.930 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:43.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:43.930 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:53.930 DEBG selecting gossips for 7a36878ac60f05645cd889671eeaba0b47cd1b0aa8f1729fd843bc20503b22f7, task: network
Oct 21 06:41:53.930 DEBG sending gossip to node 7a36878ac60f05645cd889671eeaba0b47cd1b0aa8f1729fd843bc20503b22f7, task: network
Oct 21 06:41:53.930 DEBG selecting gossips for a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 21 06:41:53.930 DEBG sending gossip to node a042da1be8d2c9feee16058d7a8c5cdab4e2f701b9d0241f4eab5a5e9ce00cb7, task: network
Oct 21 06:41:53.930 DEBG selecting gossips for d05ca82e8e41ab4a1f1a5e7723ac4c784188f800904948a60b6904e374816649, task: network
Oct 21 06:41:53.930 DEBG sending gossip to node d05ca82e8e41ab4a1f1a5e7723ac4c784188f800904948a60b6904e374816649, task: network
Oct 21 06:41:53.930 DEBG selecting gossips for e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 21 06:41:53.931 DEBG sending gossip to node e4f9ad6732d5f5c332b2ae4b81a7a39ef52b4f6b317369b01460a000835d2b0c, task: network
Oct 21 06:41:53.931 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:53.931 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:53.931 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:53.931 DEBG scheduling Read for: 0, scope: global
Oct 21 06:41:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global
Oct 21 06:41:53.931 DEBG send; frame=Data { stream_id: StreamId(7) }, scope: global

```

But it didn't receive any blocks (uptime is 110688):

```
blockRecvCnt: 0
lastBlockDate: "238.41974"
lastBlockFees: 0
lastBlockHash: 5ec4db50e6e55ae78e8ea3476adc4d06b84736d126df89225a40bd55ea232137
lastBlockHeight: "151508"
lastBlockSum: 0
lastBlockTime: ~
lastBlockTx: 0
txRecvCnt: 0
uptime: 110688
```

**Comment by mzabaluev** [comment]
@jbax There is only sending in the log excerpts you have posted, indeed.

The message of interest is "received block event". You should also receive network gossip from other nodes.

**Comment by NotToFly** [comment]
They are the same messages on repeat captured on screen shot. No block event messages.

**Comment by Micro86** [comment]
I have the same problem, must be something with the node-config.yaml file.

**Comment by smurf123444** [comment]
forward port 3000?

**Comment by Micro86** [comment]
Used other ports, original ports were open tho. 
I am now getting blocks.

**Comment by NicolasDP** [comment]
It seems this is now fixed since at least `0.7.0-rc7`.
